doc/ale-go.txt

```diff
@@ -114,5 +114,16 @@ g:ale_go_staticcheck_lint_package           *g:ale_go_staticcheck_lint_package*
   current file.
 
 
+===============================================================================
+golangserver                                              *ale-go-golangserver*
+
+g:ale_go_langserver_executable                 *g:ale_go_langserver_executable*
+                                               *b:ale_go_langserver_executable*
+  Type: |String|
+  Default: `'go-langserver'`
```

Comment:
Recent `go-langserver` supports completion with the `-gocodecompletion` flag.

Would it be possible to add this flag here and use ALE for Go completion as well? I've been successful in using it with `vim-lsp` and `asyncomplete-vim` but it would be great to be able to drop those plugins and rely on ALE 100% for linting, LSP, and completion.

I can try tackling it myself in a follow-up PR too if you want to submit this one first.

Reply:
Ok, actually took a crack at this and it works like a treat. Here is my augmented PR:

https://github.com/diegs/ale/pull/1

Feel free to add this in here, or I can rebase once you submit and send a follow-up PR.

Reply:
Added it but slightly different as now we have ale_go_langserver_options variable.